src/api-extension/cartActions/addCustomLineItemWithDiscountSummary.ts

```diff
@@ -4,13 +4,17 @@ import {
   CartActionAddCustomLineItem,
   COUPON_CUSTOM_LINE_SLUG,
 } from './CartAction';
-import couponText from './../../misc/coupon-text';
+import { ConfigService } from '@nestjs/config';
 
 export default // TODO don't create addCustomLineItem action if the summary doesn't actually change
 function addCustomLineItemWithDiscountSummary(
   cart: Cart,
   validateCouponsResult: ValidateCouponsResult,
 ): CartActionAddCustomLineItem[] {
+  const configService = new ConfigService();
+  const couponText = JSON.parse(
```

Comment:
there is a risk of misconfiguration (typo, missing closing brackets etc), leading to `Uncaught SyntaxError` when handling API Extension. It would be great to validate this configuration (array of objects with specific keys) when the application is starting